python-avd/pyavd/_eos_designs/structured_config/custom_structured_configuration/__init__.py

```diff
@@ -3,135 +3,47 @@
-            struct_cfgs.append({primary_key: item[primary_key], **struct_cfg})
-
-        return struct_cfgs
-
-    def _struct_cfg(self) -> list:
-        if struct_cfg := self.shared_utils.node_config.structured_config._as_dict():
-            return [struct_cfg]
-
-        return []
-
-    def _struct_cfgs(self) -> list:
-        if (struct_cfgs := self._hostvars.pop("struct_cfgs", None)) is not None:
-            return struct_cfgs
-
-        return []
-
-    def _ethernet_interfaces(self) -> list:
-        if (struct_cfgs := self._extract_and_apply_struct_cfg_from_list_of_dicts(self._hostvars.get("ethernet_interfaces"), "name")) == []:
-            return []
-
-        return [{"ethernet_interfaces": struct_cfgs}]
-
-    def _port_channel_interfaces(self) -> list:
-        if (struct_cfgs := self._extract_and_apply_struct_cfg_from_list_of_dicts(self._hostvars.get("port_channel_interfaces"), "name")) == []:
-            return []
-
-        return [{"port_channel_interfaces": struct_cfgs}]
-
-    def _vlan_interfaces(self) -> list:
-        if (struct_cfgs := self._extract_and_apply_struct_cfg_from_list_of_dicts(self._hostvars.get("vlan_interfaces"), "name")) == []:
-            return []
-
-        return [{"vlan_interfaces": struct_cfgs}]
-
-    def _router_bgp_peer_groups(self) -> list:
-        if self._router_bgp is None:
-            return []
-
-        if (struct_cfgs := self._extract_and_apply_struct_cfg_from_list_of_dicts(self._router_bgp.get("peer_groups"), "name")) == []:
-            return []
+    def render(self) -> None:
+        """
+        Custom Structured Configuration can contain any key, so we cannot use the regular render method.
 
-        return [
-            {
-                "router_bgp": {
-                    "peer_groups": struct_cfgs,
-                },
-            },
-        ]
+        This method merges each custom structured config on top of self.structured_config.
 
-    def _router_bgp_vrfs(self) -> list:
-        if self._router_bgp is None:
-            return []
+        First strip all (None, {}, []) from regular structured_config. This is to avoid empty objects showing up in the output dict.
+        Next we merge in custom structured config from various sources including None, {}, [].
+        """
+        # Strip empties from regular structured config
+        self.structured_config._strip_empties()
 
-        if (struct_cfgs := self._extract_and_apply_struct_cfg_from_list_of_dicts(self._router_bgp.get("vrfs"), "name")) == []:
-            return []
+        # Apply structured_config from node config
+        if struct_cfg := self.shared_utils.node_config.structured_config:
+            self.structured_config._deepmerge(struct_cfg, list_merge=self.custom_structured_configs.list_merge_strategy)
 
-        return [
-            {
-                "router_bgp": {
-                    "vrfs": struct_cfgs,
-                },
-            },
+        # Apply structured configs from base and vrfs.
```

Comment:
I would change it to the following to make more sense and not fall out of sync if this changes (even if today these are the only two):

```suggestion
        # Apply structured configs from root.
```